### ISSUE TITLE
Unable to turn on MAGE_PROFILER dependency graphs (MAGE_PROFILER 2)

### ISSUE THREAD

**Issue by duxabilii** [magento/magento2#27459]
### Preconditions (*)

1. Magento 2.3.3, Magento 2.3.4
2. Magento 2.4-develop

### Steps to reproduce (*)

1. Enable profiler dependency graphs via .htaccess adding `SetEnv MAGE_PROFILER 2`

### Expected result (*)

1. Dependency graphs displayed at the bottom of a page. 

### Actual result (*)

1. PHP Fatal Error
`Fatal error: Declaration of Magento\Framework\App\Request\PathInfo\Logger::getPathInfo(string $requestUri, string $baseUrl) must be compatible with Magento\Framework\App\Request\PathInfo::getPathInfo(string $requestUri, string $baseUrl): string in /var/www/generated/code/Magento/Framework/App/Request/PathInfo/Logger.php on line 0`

**Comment by duxabilii** [comment]
I have the same issue on `Magento dev-2.4-develop`

**Comment by perryholden** [comment]
This is interesting. Per Magento DevDocs article [Enable profiling (MAGE_PROFILER)](https://devdocs.magento.com/guides/v2.3/config-guide/bootstrap/mage-profiler.html), the numeric options are still supported, but it has to be deprecated somehow. Basically, what is happening is that when this env variable is set to "2", the `\Magento\Framework\ObjectManager\Profiler\Code\Generator\Logger` ultimately gets called to create a wrapper for each instance of a class that is instantiated by the ObjectManager, so that these classes may be profiled too.

Thing is, this generator class (again, which is used to generate the additional logger classes to be stored in `generated\code`) isn't smart enough to deal with the new return types used in PHP 7. In your specific instance, the original class wants a return type of `: string`, which the generated class does not have.

You can configure `\Magento\Framework\ObjectManager\Profiler\Code\Generator\Logger::_getMethodInfo` to take the return type into account (because `laminas/laminas-code`, which is used to generate the new logger class can generate classes with a return type), but this then in turn reveals a whole lot more that is broken.

I won't go into all the details, but the "invoke" method created by that generator class doesn't deal well with methods that return void, for instance, plus it also creates constructors that totally conflict with the classes they are overriding.

In short: this is completely broken, primarily because it's never been updated to deal with PHP 7 and their return types. I wish someone at Magento could indicate whether or not this is abandoned, or is just an oversight. Perhaps Blackfire is causing this to never be used and not be missed?

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-38977` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._
